test/functional/VM_Test/src/j9vm/runner/Runner.java

```diff
@@ -67,40 +67,46 @@
 
 	private void setPlatform() {
 		
-		String spec = System.getProperty("platform");
-		if (spec != null) {
+		String OSSpec = System.getProperty("os.name").toLowerCase();
+		if (OSSpec != null) {
 			/* Get OS from the spec string */
-			if (spec.indexOf("aix") != -1) {
+			if (OSSpec.contains("aix")) {
 				osName = OSName.AIX;
-			} else if (spec.indexOf("linux") != -1){
+			} else if (OSSpec.contains("linux")) {
 				osName = OSName.LINUX;
-			} else if (spec.indexOf("win") != -1) {
+			} else if (OSSpec.contains("windows")) {
 				osName = OSName.WINDOWS;
-			} else if (spec.indexOf("zos") != -1) {
+			} else if (OSSpec.contains("z/os")) {
 				osName = OSName.ZOS;
 			} else {
+				System.out.println("Runner couldn't determine underlying OS. Got OS Name:" + OSSpec);
 				osName = OSName.UNKNOWN;
 			}
-			
+		}
+		String archSpec = System.getProperty("os.arch").toLowerCase();
+		if (archSpec != null) {
 			/* Get arch from spec string */
-			if (spec.indexOf("ppc") != -1) {
+			if (archSpec.contains("ppc")) {
 				osArch = OSArch.PPC;
-			} else if (spec.indexOf("390") != -1) {
+			} else if (archSpec.contains("s390")) {
 				osArch = OSArch.S390X;
-			} else if (spec.indexOf("x86") != -1) {
+			} else if (archSpec.contains("amd64") || archSpec.contains("x86")) {
 				osArch = OSArch.X86;
 			} else {
+				System.out.println("Runner couldn't determine underlying architecture. Got OS Arch:" + archSpec);
 				osArch = OSArch.UNKNOWN;
 			}
 			
 			/* Get address mode */
-			if (spec.indexOf("31") != -1) {
+			if (archSpec.contains("31")) {
 				addrMode = AddrMode.BIT31;
-			} else if (spec.indexOf("32") != -1) {
+			} else if (archSpec.contains("32")) {
 				addrMode = AddrMode.BIT32;
-			} else if (spec.indexOf("64") != -1) {
+			/* os.arch on Z/OS Does not include the architecture. To support 64 bit addressing mode look for s390. */
```

Comment:
It should be `z/OS does`.
I don't see how z/OS 31-bit can be detected property. Not sure why the test uses os.arch to get the address mode, there is a better property that works across all platforms.
z/OS 31-bit `sun.arch.data.model = 32`
z/OS 64-bit `sun.arch.data.model = 64`

Reply:
Thanks for pointing that out. I've added that logic here 6a672f9e73cd5db54d843f554960c521df172d31.